### ISSUE TITLE
design: semantic boxes

### ISSUE THREAD

**Issue by elbotho** [Lodifice/mfnf-pdf-export#65]
Semantic boxes have the same indentation and blue bar as #62.
I will decide on and create the icons later.

#### Mockup
![image](https://user-images.githubusercontent.com/1258870/30439966-893d9fce-9975-11e7-8919-1b71b1613d53.png)

**Comment by elbotho** [comment]
I will probably use Icons from the Font Awesome Iconfont.
So if using another font is easier, we could just go with that.

Download at http://fontawesome.io/

**Comment by Lodifice** [comment]
This was huge, see #71. Once it is merged and the icons are ready, I will include them and close this issue.

Also note #70. It is probably the easiest to use glyphs and place them left to the space between box title and box content.

**Comment by Lodifice** [comment]
This is merged into master. No I wait for your decision on icons, @elbotho - graphics or glyphs? A dummy example would be enough to play with. Then I can close this issue and we can open a new one for missing icons (similar to #55)

**Comment by elbotho** [comment]
@Lodifice let's do it with the font. I strongly suspect we will find all the icons we need there.

**Comment by Lodifice** [comment]
Added a dummy example with FontAwesome icons for the proof environment. Now you can compare ;)

Both approaches have an issue, however.
Graphics: see #70 
Fonts: compare proofs with and without title - the placement of the symbol is fixed below the frametitle, so the two cases look different.

**Comment by elbotho** [comment]
@Lodifice Are you waiting for my input here? Or did you mean a different issue?
Is it in @kulla current upload? (otherwise I can't check it atm.)

**Comment by Lodifice** [comment]
@elbotho yes I was talking about this one. It's on the decision fonts vs graphics and if the vertical position of the font icon would be an issue.

I'll send you a beispielbuch to make sure^^

**Comment by elbotho** [comment]
Yes the placement is indeed an issue. The placement in the "Beweis ohne Titel" example is good. 
If there is no fix for this I's prefer SVG icon with the correct position.

About the LaTeX stuff: You can also ask Jonas (inyono on slack) maybe he has an idea.

**Comment by Lodifice** [comment]
We can also have a font icon placed like the SVG. It's rather "page break issue" vs. "placing issue".

So I would prefer independent decisions:
1. Font or graphics?
2. `startcode` or `startinnercode`? (and find a solution for the corresponding issue)

**Comment by elbotho** [comment]
Yeah, good separation. 

1. If there are no annoying technical problems, let's go with SVG-Icons (graphics)
2. (don't have input here)

**Comment by elbotho** [comment]
there is still discussion going on about the icons. 
but you can go ahead and create it with a standard icon, I or we can just replace the files later.

**Comment by elbotho** [comment]
Looking at the current export I have some observations. Can of course be discussed.

- [ ] The kind of box should only show on the more important boxes (e.g. Satz) or when there is a Title present. Otherwise the icons should be enough and hopefully self explanatory. 
- [ ] We should add a legend to explain the the different icons and boxes
- [ ] There should never be nested boxes. If it needed for comprehension please tell me.
- [ ] When a _Beweis_ is Part of a _Satz_ the title should not be repeated (at least if it's the same)
- [ ] Maybe we need a minimal height for boxes so the Icons don't overreach?

**Comment by Lodifice** [comment]
Switched to images in 01232e5bd0de95349f390b36813863825d17f564. There is now an icons directory with dummy pngs which you can replace individually.

However, I cannot get placements to work. If you know any Latex freaks please ask them. Otherwise I do not see icons in 1.0 (currently I commented them out)...

**Comment by elbotho** [comment]
For the minimal height I was referring to cases like this:

![image](https://user-images.githubusercontent.com/1258870/31008617-0bfbca72-a505-11e7-8ee8-263c8980e89c.png)

Not sure what kind of box that is though.

**Comment by elbotho** [comment]
New icons via 3eafbf2ec3eee1730bfe43eee647e74e2e7734f9 and  8f694ae329cae7be1191d13db6af29b9c902c90a

**Comment by Lodifice** [comment]
Regarding the minimal height: that is an important paragraph which does not have an icon.

Icons should be ok since 247b8da6f05e7b67bf84c110442e79731ccb8a9c, or do you want me to change the placing?

**Comment by Lodifice** [comment]
This has come to a happy end after 8c0a787e648e41cfa6e45d783e88d9793f0102c5 ;)
